chain/messagepool/messagepool.go

```diff
@@ -448,6 +448,36 @@ func New(ctx context.Context, api Provider, ds dtypes.MetadataDS, us stmgr.Upgra
 	return mp, nil
 }
 
+func (ms *msgSet) removeMessagesBelowBaseFee(baseFee types.BigInt) bool {
+	noncesToRemove := make([]uint64, 0)
+
+	// Find nonces of messages below the base fee and mark the first nonce to remove
+	var firstNonceToRemove uint64 = math.MaxUint64
+	for nonce, msg := range ms.msgs {
+		if types.BigCmp(msg.Message.GasFeeCap, baseFee) < 0 {
+			noncesToRemove = append(noncesToRemove, nonce)
+			if nonce > ms.nextNonce && firstNonceToRemove == math.MaxUint64 {
```

Comment:
I'm not sure I understand the `nonce > ms.nextNonce` check. Why do we only set `firstNonceToRemove` in this case?

Depending on what the intended logic we want for this, we _probably_ just want to find the lowest nonce that has `GasFeeCap < baseFee`, and then drop all subsequent messages?

Reply:
That would be the much less dumb way to do that, assuming that message sets are sorted we could just truncate.

Reply:
They're not, they're a map.